frontend/src/utils/transformationExecutor.ts

```diff
@@ -0,0 +1,101 @@
+export interface TransformationResult {
+  success: boolean;
+  document: string | object;
+  error?: string;
+  transformationId?: string;
+}
+
+// Executes a single transformation on a document
+export async function executeTransformation(
+  transformation: string,
+  document: string | object,
+): Promise<TransformationResult> {
+  try {
+    const transformFn = evaluateTransformation(transformation);
+
+    if (typeof transformFn !== "function") {
+      return {
+        success: false,
+        document,
+        error: "Transformation doesn't provide a valid entrypoint",
+      };
+    }
+
+    const result = transformFn(document);
+    return { success: true, document: result };
+  } catch (error) {
+    return {
+      success: false,
+      document,
+      error: error instanceof Error ? error.message : "Unknown error",
+    };
+  }
+}
+
+// A DocumentTransformer is a function that takes a document (object or string)
+// and returns a transformed document (object or string).
+// We need all transformations to conform to this type.
+export type DocumentTransformer = (
+  document: object | string,
+) => object | string;
```

Comment:
Rather than accept all these combinations of input/output can we expect `object` for both and make the downstream callers conform?